### PR TITLE
fix(cleaner): builder taking high CPU

### LINKED ISSUE #280: High CPU usage on GKE cluster
Builder using high CPU on GKE clusters.
Replication:
Depoly a wokrlfow-dev or workflow-beta1 charts and do a docker stats on the builder container:
```
CONTAINER           CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O
567dcf2a8bbe        96.17%              40.31 MB / 3.892 GB   1.04%               0 B / 0 B           991.2 kB / 3.883 MB
```
top on the node gives
```
top - 18:13:25 up 23:35,  1 user,  load average: 1.10, 1.31, 1.24
Tasks: 106 total,   1 running, 105 sleeping,   0 stopped,   0 zombie
%Cpu(s): 99.0 us,  1.0 sy,  0.0 ni,  0.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
KiB Mem:   3801024 total,  3286940 used,   514084 free,   383544 buffers
KiB Swap:        0 total,        0 used,        0 free,  1993720 cached

  PID USER      PR  NI  VIRT  RES  SHR S  %CPU %MEM    TIME+  COMMAND                                                                                                                                                                                                                                                                                                       
17835 root      20   0 26220  22m    4 S  96.7  0.6   1251:42 boot                                                                                                                                                                                                                                                                                                          
 3310 root      20   0  232m  76m  30m S   0.7  2.1  16:02.86 kubelet                                                                                                                                                                                                                                                                                                       
 4241 root      20   0  739m 164m 8136 S   0.7  4.4  11:48.61 google-fluentd                                                                                                                                                                                                                                                                                                
 3872 root      20   0 1348m  48m  18m S   0.3  1.3  14:47.96 docker                                                                                                                                                                                                                                                                                                        
17912 root      20   0 65384  49m  13m S   0.3  1.3   0:58.64 registry     
```

### PR DESCRIPTION
**Pull request by kmala**
# Summary of Changes

This PR fixes the issue builder taking high CPU by waiting on the channel stream using select.

# Issue(s) that this PR Closes

Fixes #280 

# Testing Instructions
After the fix:
```
CONTAINER           CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O
5be5bad7ce1a        0.32%               22.45 MB / 3.892 GB   0.58%               0 B / 0 B           0 B / 0 B
```
Testing: 

1. Create a Deis Cluster
2. Deploy an App
3. Destroy the App
4. Check the /home/git/ folder and it shouldn't contain the created app git repo

# Pull Request Hygiene TODOs

Please make sure the below checklist is complete.

- [x] Your pull request is concise and to the point (make another PR for refactoring nearby code)
- [x] Your commits are squashed into logical units of work
- [x] Your commits follow the [commit style guidelines](http://docs.deis.io/en/latest/contributing/standards.html#commit-style)

### CODE REVIEW

**Comment by arschles** [comment]
@kmala *great* catch, and I take full responsibility for making the mistake in the first place (doh!) I've marked LGTM, and checked off the boxes above (under `Pull Request Hygiene TODOs` - in the future, can you check those off when they're done?)

Thanks!

**Comment by arschles** [comment]
@kmala I see in #280, @MaxenceAdnot says the following: `I noticed that a defunct child process linked to boot was consuming a lot of CPU`. I agree that this PR cleans up the wait logic a bit, but does it solve the defunct process issue?

File: pkg/cleaner/cleaner.go [hunk 1]
```diff
@@ -26,16 +26,18 @@ func Run(gitHome string, nsLister k8s.NamespaceWatcher, fs sys.FS) error {
...
-		event := <-watcher.ResultChan()
-		if event.Type == "DELETED" {
-			switch event.Object.(type) {
-			case (*api.Namespace):
-				namespace := event.Object.(*api.Namespace)
-				appToDelete := gitHome + "/" + namespace.ObjectMeta.Name + dotGitSuffix
-				if err := fs.RemoveAll(appToDelete); err != nil {
-					log.Printf("Cleaner error removing deleted app %s (%s)", appToDelete, err)
+		select {
+		case event := <-watcher.ResultChan():
```
Comment (arschles) [inline pkg/cleaner/cleaner.go]: How does the logic here change from the above `event := <-watcher.ResultChan()` wait?

Also, as a defensive measure, can you check for channel close here and exit the loop if so?

**Comment by arschles** [comment]
@kmala I added a few comments just now, so removed my LGTM. Sorry for the flip-flopping here.

**Comment by kmala** [comment]
i don't see any defunct process attached to the boot. If you see the issue description in #280  the boot is the one which clearly is taking most of the CPU.

**Comment by arschles** [comment]
@kmala ok, fair enough. let's keep an eye out for the defunct child process issue then

**Comment by MaxenceAdnot** [comment]
I'll try to reproduce the case with the defunct child process on my current cluster and I'll let you know. The root cause of these two issues could be different.

**Comment by kmala** [comment]
@MaxenceAdnot Yes...your defunct process are from git push failure

**Pull request merged**